database/postgresDB.go

```diff
@@ -0,0 +1,22 @@
+package database
+
+import (
+	"context"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+func NewPostgreSQLConnection(uri string, connections int32) (*pgxpool.Pool, error) {
+	config, err := pgxpool.ParseConfig(uri)
+	if err != nil {
+		return nil, err
+	}
+	config.MaxConns = connections
+	//create a new connection pool
+	pool, err := pgxpool.ConnectConfig(context.Background(), config)
+	if err != nil {
+		return nil, err
+	}
+	PostgreDB = pool
```

Comment:
Same issue of garbage collection and memory usage...do note this and try to avoid going forward in all areas of code.

Also notice that it's `PostgresSQLDB` and not `PostgreDB` do change this everywhere.
```go
PostgresSQLDB, err := pgxpool.ConnectConfig(context.Background(), config)
if err != nil {
	return nil, err
}
```